docs/developer/HootenannyValidation.asciidoc

```diff
@@ -8,68 +8,67 @@ https://josm.openstreetmap.de/[JOSM] via the
+public Maven JOSM repository), you must manually install it with Maven. See 
+`scripts/core/make-hoot-josm-java.sh` for an example.
+* Within `hoot-josm/pom.xml`. Under the `<properties>` XML tag, configure `josm.artifactId` and 
+`josm.version` for the JOSM JAR you wish to use. This will use the configured Maven repository to 
+locate and download the JAR file.
+* The JAR file will exist in the Maven cache at: 
+`~/.m2/repository/org/openstreetmap/josm/josm/<josm.version>/josm-<version>.jar. Locating the JAR 
+file in the cache can be useful for debugging purposes.
+* At build time the JOSM jar will be copied to `hoot-josm/target/dependency-jars/josm.jar`, and its 
+path referenced within the `jni.class.path` configuration option. Note that `hoot-josm` will modify
+the JAR file via a monkey patch. This is necessary to get around some compatibility issues. See 
+`scripts/core/make-hoot-josm-java.sh` and the implementation of `AbstractProperty` in hoot-josm for 
+more detail.
+* Hootenanny produces its own JAR file to interact with the JOSM JAR. By default, this file will be 
+copied to `hoot-josm/target/hoot-josm.jar`.
+* The value of the configuration option, `josm.validators`, must match the expected validators
+available in the JAR. By default the option is configured to work with public JOSM. Other versions
+may use different validators.
 
-* Check out a branch of code without the changes that need to be evaluated to be used as a baseline 
-(e.g. master branch). 
-* Run selected tests on the baseline branch.
-* Generate a test validation report for the baseline branch using `test-output` as the input 
-directory.
-* Switch to another source directory and check out the branch of your code with the changes that 
-need to be evaluated.
-* Run the same tests as in the previous steps on the second test branch.
-* Generate a test validation report on the second test branch using `test-output` as the input 
-directory.
-* Finally, use the `diff` tool on the two validation reports to determine the difference in 
-generated validation errors before and after your changes.
+See the jni.* and josm.* configuration options for more detail on JOSM configuration.
 
-Here is a complete example:
------
-cd hoot
-
-git checkout master
-aclocal && autoconf && autoheader && automake --add-missing --copy && ./configure --with-rnd --with-josm --with-services
-make -j$(nproc)
-HootTest --glacial --parallel
-hoot validate test-output --recursive "*" --report-output=tmp/master-validation-report
-
-git checkout my-feature-branch
-aclocal && autoconf && autoheader && automake --add-missing --copy && ./configure --with-rnd --with-josm --with-services
-make -j$(nproc)
-HootTest --glacial --parallel
-hoot validate test-output --recursive "*" --report-output=tmp/my-feature-branch-validation-report
------
+JNI is notoriously difficult to debug and its usage with JOSM can become very brittle with 
```

Comment:
This is just a standard disclaimer that upgrading JOSM may not be a trivial task due to JNI only having runtime checks and no compile time checks. Upgrade with caution.